data-prepper-plugins/otel-trace-raw-processor/src/main/java/com/amazon/dataprepper/plugins/processor/oteltrace/OTelTraceRawProcessor.java

```diff
@@ -39,14 +41,14 @@
     private static final BigDecimal SEC_TO_MILLIS = new BigDecimal(1_000);
 
     private static final Logger log = LoggerFactory.getLogger(OTelTraceRawProcessor.class);
+    private final Counter spanCounter;
+    private final Counter resourceSpanCounter;
 
     //TODO: https://github.com/opendistro-for-elasticsearch/simple-ingest-transformation-utility-pipeline/issues/66
     public OTelTraceRawProcessor(final PluginSetting pluginSetting) {
-        this();
-    }
-
-    private OTelTraceRawProcessor() {
-
+        super(pluginSetting);
+        spanCounter = this.pluginMetrics.counter("spanCounter");
```

Comment:
1. Lets keep metric names in static final variables
2. Span counter doesn't really capture what this metric is doing. This is counting the number of *errors* encountered when processing individual spans

Reply:
Agreed the naming "spanCounter" is confusing as I review the doExecute.